### PR TITLE
Auth screens and functionality

### PR DESCRIPTION
**Pull request by pankaj-prog**
feat: auth screens
feat: Axios custom hook for API calls
feat: authentication functionality

### CODE REVIEW

File: src/components/UserRoutes/UserRoutes.jsx [hunk 1]
```diff
@@ -1,8 +1,16 @@
 import React from "react";
-import { Outlet } from "react-router-dom";
+import { Outlet, useLocation, Navigate } from "react-router-dom";
+import { useAuth } from "../../context";
```
Comment (Debarshi95) [inline src/components/UserRoutes/UserRoutes.jsx:3]: Use absolute imports

File: src/pages/auth/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,165 @@
+import React, { useRef, useState, useEffect } from "react";
+import "./auth.css";
+
+import { Link, useLocation, useNavigate } from "react-router-dom";
+import { useAxios } from "../../utils/useAxios";
+import { useAuth } from "../../context";
```
Comment (Debarshi95) [inline src/pages/auth/Login.jsx]: Same here. Avoid    ../../

File: src/pages/auth/Signup.jsx [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+import {
+  validateName,
+  validateEmail,
+  validatePassword,
+} from "./utils/inputValidateUtils";
+import { useAuth } from "../../context";
+import { useAxios } from "../../utils/useAxios";
+
+const SignUp = () => {
+  const initalFormValidateState = {
```
Comment (Debarshi95) [inline src/pages/auth/Signup.jsx:14]: You can move this object out of the component body. That will it won't be recreated on each re render

**Review by Debarshi95 — APPROVED** [review]
Good work the UI and author functionalities. 
Added some minor comments. May be work on them.

File: src/pages/auth/Login.jsx [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    isEmailValid: true,
+    isPasswordValid: true,
+  };
+  const [formValidateState, setFormValidateState] = useState(
+    initialFormValidateState
+  );
+
+  const emailRef = useRef();
+  const passwordRef = useRef();
+
```
Comment (m26unkwn) [inline src/pages/auth/Login.jsx:22]: like that you are using ref for the forms.

**Review by m26unkwn — APPROVED** [review]
You can start working on other things.
You can also look up for the other folks comment

**Pull request merged**